test/jdk/java/awt/dnd/DragSourceMotionListenerTest.java

```diff
@@ -0,0 +1,250 @@
+
+    private static final DragSourceAdapter sourceAdapter = new DragSourceAdapter() {
+        public void dragMouseMoved(DragSourceDragEvent dsde) {
+            if (Math.abs(dsde.getX() - testPoint1.getX()) < 5) {
+                passedTest1 = true;
+            }
+        }
+
+        public void dragDropEnd(DragSourceDropEvent dsde) {
+            if (Math.abs(dsde.getX() - testPoint2.getX()) < 5) {
+                passedTest2 = true;
+            }
+        }
+    };
+
+    private static final DropTargetListener targetAdapter = new DropTargetAdapter() {
+        public void drop(DropTargetDropEvent e) {
+            e.acceptDrop(DnDConstants.ACTION_COPY);
+            try {
+                final Transferable t = e.getTransferable();
+                final String str =
+                        (String) t.getTransferData(DataFlavor.stringFlavor);
+                e.dropComplete(true);
+            } catch (Exception ex) {
+                ex.printStackTrace();
+                e.dropComplete(false);
+            }
+        }
+    };
+
+    private static final DropTarget dropTarget = new DropTarget(target, targetAdapter);
+    Component clickedComponent = null;
+
+    private void createAndShowUI() {
+        frame = new Frame("DragSourceMotionListenerTest");
+        ds.addDragSourceListener(sourceAdapter);
+        ds.addDragSourceMotionListener(sourceAdapter);
+        ds.createDefaultDragGestureRecognizer(source, DnDConstants.ACTION_COPY, gestureListener);
+        target.setDropTarget(dropTarget);
+
+        frame.setLayout(new GridLayout(1, 2));
+
+        frame.add(source);
+        frame.add(target);
+
+        Toolkit.getDefaultToolkit()
+               .addAWTEventListener(this, AWTEvent.MOUSE_EVENT_MASK);
+        frame.pack();
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        final Dimension[] d = new Dimension[1];
+        try {
+            Robot robot = new Robot();
+            robot.setAutoDelay(10);
+
+            DragSourceMotionListenerTest dsmObj = new DragSourceMotionListenerTest();
+            EventQueue.invokeAndWait(dsmObj::createAndShowUI);
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            EventQueue.invokeAndWait(() -> {
+                srcPoint = source.getLocationOnScreen();
+                d[0] = source.getSize();
+                srcPoint.translate(d[0].width / 2, d[0].height / 2);
+            });
+            robot.waitForIdle();
+
+            if (!dsmObj.pointInComponent(robot, srcPoint, source)) {
+                throw new RuntimeException("WARNING: Couldn't locate source panel.");
+            }
+
+            EventQueue.invokeAndWait(() -> {
+                dstOutsidePoint = frame.getLocationOnScreen();
+                d[0] = frame.getSize();
+                dstOutsidePoint.translate(3 * d[0].width / 2, d[0].height / 2);
+                testPoint1.setLocation(dstOutsidePoint);
```

Comment:
The same issue as above.
```suggestion
                Point p = frame.getLocationOnScreen();
                Dimension d = frame.getSize();
                p.translate(3 * d.width / 2, d.height / 2);
                dstOutsidePoint = p;
                testPoint1.setLocation(p);
```